service/test/test_sst_priority.sh

```diff
@@ -0,0 +1,106 @@
+#!/bin/bash
+#  Copyright (c) 2015 - 2021, Intel Corporation
+#
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions
+#  are met:
+#
+#      * Redistributions of source code must retain the above copyright
+#        notice, this list of conditions and the following disclaimer.
+#
+#      * Redistributions in binary form must reproduce the above copyright
+#        notice, this list of conditions and the following disclaimer in
+#        the documentation and/or other materials provided with the
+#        distribution.
+#
+#      * Neither the name of Intel Corporation nor the names of its
+#        contributors may be used to endorse or promote products derived
+#        from this software without specific prior written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+#  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+#  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+#  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+#  OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+#  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+#  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+#  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+#  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+#  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY LOG OF THE USE
+#  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+if [[ $# -gt 0 ]] && [[ $1 == '--help' ]]; then
+    echo "
+    Set Core Priority with SST
+    --------------------------
+
+    Runs a CPU stress test binary (stress-ng) on all cores of the
+    platform.  While the all of the CPUs are loaded the GEOPM service
+    is used to configure the Intel SST feature to give high priority
+    to two cores (cores 0 and 5).  The CPU frequencies of the first 8
+    cores on the system are printed before and after configuring SST
+    core priority.
+
+    This test is a work in progress.
+
+"
+    exit 0
+fi
+
+# Input for geopmsession to read frequency
+
+# Input for geopmsession to configure SST
+BENCH_LOG=$(mktemp)
+SESSION_LOG=$(mktemp)
+READ_REQUEST=$(mktemp)
+WRITE_REQUEST=$(mktemp)
+
+echo 'CPU_FREQUENCY_STATUS core 0
+CPU_FREQUENCY_STATUS core 1
+CPU_FREQUENCY_STATUS core 2
+CPU_FREQUENCY_STATUS core 3
+CPU_FREQUENCY_STATUS core 4
+CPU_FREQUENCY_STATUS core 5
+CPU_FREQUENCY_STATUS core 6
+CPU_FREQUENCY_STATUS core 7' > ${READ_REQUEST}
+
+echo 'SST::COREPRIORITY_ENABLE:ENABLE board 0 1
```

Comment:
Sorry, I omitted some steps in the runs I did earlier, since I had already configured them earlier in that session. You'll also want to set each of the weights and the frequency limits.

For buckets 0 through 3, something like:
```
SST::COREPRIORITY:<bucket>:WEIGHT board 0 <weight>
SST::COREPRIORITY:<bucket>:FREQUENCY_MIN board 0 <the starting frequency>
SST::COREPRIORITY:<bucket>:FREQUENCY_MAX board 0 <e.g. the system's max frequency>
```
Note that the buckets assigned here are defining the behavior that you map to cores with `SST::COREPRIORITY:ASSOCIATION`. That is, the `<bucket>` here is selected for a core by using that number as the input to `SST::COREPRIORITY:ASSOCIATION`.

For one configuration that would work for this test, you could set the min frequency to sticker for all except bucket 0. Maybe set bucket 0 to start somewhere inside the turbo range. Set a increasing weight as the bucket increases. You might consider bucket 0->0 weight, bucket 1->5 weight, bucket 2->10 weight, bucket 3->15 weight.